### PR TITLE
Added randomness to AI's decision which influence to reveal.

### PR DESCRIPTION
**Pull request by einsteinsfool**
Before this commit, AI's first revealed influence tells us a lot about the second one.
E.g. if it reveals a duke, we're 100% sure that the other card is duke.
If it reveals an assassin, then the other card is either a duke or also an assassin.
Etc.

Now, AI will still probably choose to reveal ambassador over duke but we're not 100% sure.

Feel free to modify rank probabilities as you please.
All tests pass.

### CODE REVIEW

File: ai-player.js [hunk 1]
```diff
@@ -525,28 +528,28 @@ function createAiPlayer(game, options) {
...
+        if (influence.length > 1) {
+            var influenceProbability = [];
+            for(var i = 0; i < influence.length; i++) {
+                for(var j = 0; j < roleWeights[influence[i]]; j++) {
+                    influenceProbability.push(i);
                 }
-                return;
             }
+            chosenInfluence = influenceProbability[rand.intBetween(0, influenceProbability.length-1)];
+            influenceProbability = null;
```
Comment (octachrome) [inline ai-player.js]: Nit: this line does nothing

File: ai-player.js [hunk 2]
```diff
@@ -525,28 +528,28 @@ function createAiPlayer(game, options) {
...
-                    role: role
-                });
-                // Don't claim this role any more.
-                if (claims[state.playerIdx]) {
-                    delete claims[state.playerIdx][role];
+        var chosenInfluence = 0;
+
+        if (influence.length > 1) {
+            var influenceProbability = [];
+            for(var i = 0; i < influence.length; i++) {
```
Comment (octachrome) [inline ai-player.js]: Nit: the convention used elsewhere in the source is for a space between `for` and `(`

**Comment by octachrome** [comment]
This looks fine. I raised a couple of very minor points, but there's no need to fix them in this pull request.

**Comment by einsteinsfool** [comment]
I'll add spaces in the next PR, sorry.
As for the nullifying, I thought it will allow garbage collector to dispose the array.
But I don't normally write in JS, so I'm not sure. Just what I read on SO.
Should I remove it in the next PR?

**Comment by octachrome** [comment]
> I thought it will allow garbage collector to dispose the array

`influenceProbablility` is a local variable, so as soon as the function returns it will go out of scope and garbage collection can occur.

You only need to worry about nulling a reference when it is a global variable and will live for a long time.

**Pull request merged**